### DISCUSSION TITLE
multiple background tasks showing not reliable behavior

### DISCUSSION THREAD

**Discussion by jkh911208** [fastapi/fastapi#7167] [Questions] [answered]
### First check

* [x] I added a very descriptive title to this issue.
* [xI used the GitHub search to find a similar issue and didn't find it.
* [ ] I searched the FastAPI documentation, with the integrated search.
* [x] I already searched in Google "How to X in FastAPI" and didn't find any information.
* [x] I already read and followed all the tutorial in the docs and didn't find an answer.
* [x] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/samuelcolvin/pydantic).
* [x] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
* [x] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).
* [x] After submitting this, I commit to one of:
    * Read open issues with questions until I find 2 issues where I can help someone and add a comment to help there.
    * I already hit the "watch" button in this repository to receive notifications and I commit to help at least 2 people that ask questions in the future.
    * Implement a Pull Request for a confirmed bug.

### Example

```Python
import asyncio
import logging

from fastapi import FastAPI
from fastapi.responses import JSONResponse
from starlette.background import BackgroundTask, BackgroundTasks

app = FastAPI()

tasks = BackgroundTasks()


@app.get("/")
async def read_root():
    response_object = {"Hello": "World"}
    tasks.add_task(bg_task)
    return JSONResponse(response_object, 202, background=tasks)


async def bg_task():
    logging.error("test1")
    await asyncio.sleep(10)
    logging.error("test2")

if __name__ == "__main__":
    import uvicorn
    uvicorn.run("main:app", host="0.0.0.0", port=8000,
                log_level="debug", reload=True)

```

### Description

go to open api ui localhost:8000/docs

make request to root route you will see "test1" and then after 10 seconds you will see "test2"
which is working as expected, but if you make request and before it print out "test2" make another request.
then background task will print out bunch of "test1" and "test2" randomly

### Environment

* OS: Debian
* FastAPI Version [e.g. 0.3.0]: 0.61.1
* Python version: 3.8

### Additional context

<img width="516" alt="Screen Shot 2020-10-20 at 7 34 12 PM" src="https://user-images.githubusercontent.com/17028674/96659413-7ba6db00-130c-11eb-9aa8-3c652b4cdebd.png">

<img width="551" alt="Screen Shot 2020-10-20 at 7 35 28 PM" src="https://user-images.githubusercontent.com/17028674/96659420-806b8f00-130c-11eb-846a-0ac835b4190a.png">

**Comment by longxiaofei** [comment, marked as answer]
```
import asyncio
import logging

from fastapi import FastAPI
from fastapi.responses import JSONResponse
from starlette.background import BackgroundTask, BackgroundTasks

app = FastAPI()


@app.get("/")
async def read_root(tasks: BackgroundTasks):
    response_object = {"Hello": "World"}
    tasks.add_task(bg_task)
    return JSONResponse(response_object, 202, background=tasks)


async def bg_task():
    logging.error("test1")
    await asyncio.sleep(10)
    logging.error("test2")

if __name__ == "__main__":
    import uvicorn
    uvicorn.run("main:app", host="0.0.0.0", port=8000,
                log_level="debug", reload=True)
```

You need to create a new instance(BackgroundTasks) every time you receive a request.

**Comment by jkh911208** [comment]
```python
import asyncio
import logging

from fastapi import FastAPI
from fastapi.responses import JSONResponse
from starlette.background import BackgroundTask, BackgroundTasks

app = FastAPI()


@app.get("/")
async def read_root():
    response_object = {"Hello": "World"}
    tasks = BackgroundTasks()
    tasks.add_task(bg_task)
    return JSONResponse(response_object, 202, background=tasks)


async def bg_task():
    logging.error("test1")
    await asyncio.sleep(10)
    logging.error("test2")

if __name__ == "__main__":
    import uvicorn
    uvicorn.run("test:app", host="0.0.0.0", port=8000,
                log_level="debug", reload=True)
```

@longxiaofei thank you, above code worked to meet my expectation

**Comment by jkh911208** [comment]
this issue is resolved. by creating the instance every time it is been called

**Comment by tiangolo** [comment]
Thanks for the help here, everyone! And thanks @jkh911208 for closing the issue.

> Sorry for the long delay! 🙈  I wanted to personally address each issue/PR and they piled up through time, but now I'm checking each one in order.
